JoyboyCommunity/src/modules/TipModal/index.tsx

```diff
@@ -45,6 +46,26 @@ export const TipModal = forwardRef<Modalize, TipModalProps>(
 
     const {showDialog, hideDialog} = useDialog();
 
+    useEffect(() => {
+      const handleChainIdChange = () => {
+        const chainId = account.chainId;
+        const currentChainId = NETWORK_NAME === 'SN_MAIN' ? mainnet.id : sepolia.id;
```

Comment:
You can use [`useNetwork hook`](https://starknet-react.com/hooks/useNetwork) with `starknetChainId` to get the chain id and check against CHAIN_ID.

```ts
import { starknetChainId, useNetwork } from '@starknet-react/core';

const { chain } = useNetwork();

useEffect(() => {
  const chainId = chain.id ? starknetChainId(chain.id) : undefined;
  
  if (chainId) {
    if (chainId !== CHAIN_ID) {
      // show dialog
    } else {
      // hide dialog
    }
  }
}, [chain.id])
```